src/main/java/mate/academy/spring/controller/ShoppingCartController.java

```diff
@@ -25,23 +27,30 @@ public class ShoppingCartController {
     public ShoppingCartController(ShoppingCartService shoppingCartService,
                                   UserService userService,
                                   MovieSessionService movieSessionService,
-            ResponseDtoMapper<ShoppingCartResponseDto, ShoppingCart>
-                                      shoppingCartResponseDtoMapper) {
+                                  ResponseDtoMapper<ShoppingCartResponseDto, ShoppingCart>
+                                          shoppingCartResponseDtoMapper) {
         this.shoppingCartService = shoppingCartService;
         this.userService = userService;
         this.movieSessionService = movieSessionService;
         this.shoppingCartResponseDtoMapper = shoppingCartResponseDtoMapper;
     }
 
     @PutMapping("/movie-sessions")
-    public void addToCart(@RequestParam Long userId, @RequestParam Long movieSessionId) {
+    public void addToCart(Authentication authentication, @RequestParam Long movieSessionId)
+            throws AuthenticationException {
+        User user = userService.findByEmail(authentication.getName())
+                .orElseThrow(() -> new AuthenticationException("Can't find user with this email: "
```

Comment:
Actually it's not an appropriate type of exception here. Please fix the same issue in all places